python/ray/dag/compiled_dag_node.py

```diff
@@ -858,16 +881,29 @@ def _get_or_compile(
             if isinstance(task.dag_node, ClassMethodNode):
                 # `readers` is the nodes that are ordered after the current one (`task`)
                 # in the DAG.
-                readers = [self.idx_to_task[idx] for idx in task.downstream_node_idxs]
+                # readers = [self.idx_to_task[idx] for idx in task.downstream_node_idxs]
+                readers = []
+                for idx in task.downstream_node_idxs:
+                    downstream_task = self.idx_to_task[idx]
+                    readers.append(downstream_task)
```

Comment:
Nit: Skip adding TaskReturnNode here instead of filtering out the TaskReturnNodes later on.